### PR TITLE
Flink 5386 pr inner2

### PR DESCRIPTION
**Pull request by sunjincheng121**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [ ] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [ ] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [ ] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

File: docs/dev/table_api.md [hunk 1]
```diff
@@ -1022,69 +1022,70 @@ Temporal intervals can be represented as number of months (`Types.INTERVAL_MONTH
...
+The Table API is a declarative API to define queries on batch and streaming tables. Projection, selection, and union operations can be applied both on streaming and batch tables without additional semantics. Aggregations on (possibly) infinite streaming tables, however, can only be computed on finite groups of records. Window aggregates group rows into finite groups based on time or row-count intervals and evaluate aggregation functions once per group. For batch tables, windows are a convenient shortcut to group records by time intervals.
 
-Group-windows are defined using the `window(w: GroupWindow)` clause. The following example shows how to define a group-window aggregation on a table.
+Windows are defined using the `window(w: Window)` clause, when define a window it must be assigned the window an alias using `as`. Window aggregates must with additional `groupBy` attribute, it should be noted that the window alias must be included in `groupBy` statement.
```
Comment (shaoxuan-wang) [inline docs/dev/table_api.md]: Windows are defined using the `window(w: Window)` clause. When defining the window, it must assign an alias to the window, using `as`.  It should be noted that in window aggregation, the window alias must be referenced in an additional `groupBy` attribute. The following example shows how to define a window aggregation on a table.

File: docs/dev/table_api.md [hunk 2]
```diff
@@ -1022,69 +1022,70 @@ Temporal intervals can be represented as number of months (`Types.INTERVAL_MONTH
...
+  .groupBy("w") 
   .select("b.sum")        // aggregate
 {% endhighlight %}
 </div>
 
 <div data-lang="scala" markdown="1">
 {% highlight scala %}
 val table = input
-  .window(w: GroupWindow) // define window
+  .window(window as 'w) // define window with alias
```
Comment (shaoxuan-wang) [inline docs/dev/table_api.md]: [w: Window] as 'w

File: docs/dev/table_api.md [hunk 3]
```diff
@@ -1022,69 +1022,70 @@ Temporal intervals can be represented as number of months (`Types.INTERVAL_MONTH
...
 
-Group-windows are defined using the `window(w: GroupWindow)` clause. The following example shows how to define a group-window aggregation on a table.
+Windows are defined using the `window(w: Window)` clause, when define a window it must be assigned the window an alias using `as`. Window aggregates must with additional `groupBy` attribute, it should be noted that the window alias must be included in `groupBy` statement.
 
 <div class="codetabs" markdown="1">
 <div data-lang="java" markdown="1">
 {% highlight java %}
 Table table = input
-  .window(GroupWindow w)  // define window
+  .window(window.as("w"))  // define window with alias
```
Comment (shaoxuan-wang) [inline docs/dev/table_api.md]: (Window w).as("w")

File: docs/dev/table_api.md [hunk 4]
```diff
@@ -1022,69 +1022,70 @@ Temporal intervals can be represented as number of months (`Types.INTERVAL_MONTH
...
-  .window(w: GroupWindow) // define window
+  .window(window as 'w) // define window with alias
+  .groupBy('w) 
   .select('b.sum)         // aggregate
 {% endhighlight %}
 </div>
 </div>
 
-In streaming environments, group-window aggregates can only be computed in parallel, if they are *keyed*, i.e., there is an additional `groupBy` attribute. Group-window aggregates without additional `groupBy`, such as in the example above, can only be evaluated in a single, non-parallel task. The following example shows how to define a keyed group-window aggregation on a table. 
+In streaming environments, window aggregates can only be computed in parallel, if they are *keyed*, i.e., such as in the example above, can only be evaluated in a single, non-parallel task. The following example shows how to define a keyed window aggregation on a table.
```
Comment (shaoxuan-wang) [inline docs/dev/table_api.md]: In streaming environments, window aggregates can only be computed in parallel, if they are *keyed*, i.e., besides the window alias there are additional grouping keys in `groupBy`. Window aggregates without grouping keys, such as in the example above, can only be evaluated in a single, non-parallel task. The following example shows how to define a keyed group-window aggregation on a table.

File: docs/dev/table_api.md [hunk 5]
```diff
@@ -1022,69 +1022,70 @@ Temporal intervals can be represented as number of months (`Types.INTERVAL_MONTH
...
-In streaming environments, group-window aggregates can only be computed in parallel, if they are *keyed*, i.e., there is an additional `groupBy` attribute. Group-window aggregates without additional `groupBy`, such as in the example above, can only be evaluated in a single, non-parallel task. The following example shows how to define a keyed group-window aggregation on a table. 
+In streaming environments, window aggregates can only be computed in parallel, if they are *keyed*, i.e., such as in the example above, can only be evaluated in a single, non-parallel task. The following example shows how to define a keyed window aggregation on a table. 
 
 <div class="codetabs" markdown="1">
 <div data-lang="java" markdown="1">
 {% highlight java %}
 Table table = input
-  .groupBy("a")
-  .window(GroupWindow w)  // define window
+  .window(window.as("w"))  // define window with alias
```
Comment (shaoxuan-wang) [inline docs/dev/table_api.md]: (Window w).as("w")

File: docs/dev/table_api.md [hunk 6]
```diff
@@ -1022,69 +1022,70 @@ Temporal intervals can be represented as number of months (`Types.INTERVAL_MONTH
...
   .select("a, b.sum")     // aggregate
 {% endhighlight %}
 </div>
 
 <div data-lang="scala" markdown="1">
 {% highlight scala %}
 val table = input
-  .groupBy('a)
-  .window(w: GroupWindow) // define window
+  .window(window as 'w) // define window with alias
```
Comment (shaoxuan-wang) [inline docs/dev/table_api.md]: (Window w).as("w")

File: docs/dev/table_api.md [hunk 7]
```diff
@@ -1022,69 +1022,70 @@ Temporal intervals can be represented as number of months (`Types.INTERVAL_MONTH
...
+  .groupBy('w, 'a)
   .select('a, 'b.sum)     // aggregate
 {% endhighlight %}
 </div>
 </div>
 
-The `GroupWindow` parameter defines how rows are mapped to windows. `GroupWindow` is not an interface that users can implement. Instead, the Table API provides a set of predefined `GroupWindow` classes with specific semantics, which are translated into underlying `DataStream` or `DataSet` operations. The supported window definitions are listed below. 
-By assigning the group-window an alias using `as`, properties such as the start and end timestamp of a time window can be accessed in the `select` statement.
+The `Window` parameter defines how rows are mapped to windows. `Window` is not an interface that users can implement. Instead, the Table API provides a set of predefined `Window` classes with specific semantics, which are translated into underlying `DataStream` or `DataSet` operations. The supported window definitions are listed below. Window properties such as the start and end timestamp of a time window can be accessed in the `select` statement.
```
Comment (shaoxuan-wang) [inline docs/dev/table_api.md]: Window properties such as the start and end timestamp of a time window can be accessed in the `select` statement via window alias.

File: docs/dev/table_api.md [hunk 8]
```diff
@@ -1022,69 +1022,70 @@ Temporal intervals can be represented as number of months (`Types.INTERVAL_MONTH
...
 
 <div class="codetabs" markdown="1">
 <div data-lang="java" markdown="1">
 {% highlight java %}
 Table table = input
-  .groupBy("a")
-  .window(XXX.as("myWin"))                      // define window alias
-  .select("a, myWin.start, myWin.end, b.count") // aggregate
+  .window(window.as("w"))                      // define window with alias
```
Comment (shaoxuan-wang) [inline docs/dev/table_api.md]: .window((Window w).as("w"))

File: docs/dev/table_api.md [hunk 9]
```diff
@@ -1022,69 +1022,70 @@ Temporal intervals can be represented as number of months (`Types.INTERVAL_MONTH
...
 {% endhighlight %}
 </div>
 
 <div data-lang="scala" markdown="1">
 {% highlight scala %}
 val table = input
-  .groupBy('a)
-  .window(XXX as 'myWin)                          // define window alias
-  .select('a, 'myWin.start, 'myWin.end, 'b.count) // aggregate
+  .window(window as 'w)                          // define window with alias
```
Comment (shaoxuan-wang) [inline docs/dev/table_api.md]: [w: Window] as 'w

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/table.scala [hunk 10]
```diff
@@ -795,14 +795,14 @@ class Table(
     * For batch tables of finite size, windowing essentially provides shortcuts for time-based
     * groupBy.
     *
-    * __Note__: window on non-grouped streaming table is a non-parallel operation, i.e., all data
-    * will be processed by a single operator.
-    *
-    * @param groupWindow group-window that specifies how elements are grouped.
+    * @param window window that specifies how elements are grouped.
```
Comment (shaoxuan-wang) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/table.scala]: 加上：
 __Note__: window without grouping keys streaming table is a non-parallel operation, i.e., all data	will be processed by a single operator.

File: docs/dev/table_api.md (hunk 4 above)
Comment (sunjincheng121) [inline docs/dev/table_api.md]: In streaming environments, window aggregates can only be computed in parallel, if they are keyed, i.e., besides the window alias there are additional grouping keys in groupBy. Window aggregates only with window alias, such as in the example above, can only be evaluated in a single, non-parallel task. The following example shows how to define a keyed window group aggregation on a table.

**Pull request closed without merging**